### PR TITLE
Fix more lane icon edge cases and tests

### LINKED ISSUE #3107: Reenable LaneViewTests
LaneViewTests was excluded from the MapboxNavigationTests target that #3043 revived via SPM, due to numerous assertion failures. At a glance, it appears to be a mix of copypasta from #2882 (which was merged while MapboxNavigationTests was still kaput) and some potential bugs in the SDK itself.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by 1ec5**
This PR fixes several edge cases in lane indication ranking that I missed in #2882 due to disabled unit tests. It also fixes numerous errors in the tests themselves. In some cases, the library or tests weren’t wrong, but the additional assets in #2882 make a better result possible. LaneViewTests has been reenabled, now that it passes.

Fixes #3107.

/cc @mapbox/navigation-ios @avi-c

### CODE REVIEW

File: Sources/MapboxNavigation/LaneView.swift [hunk 1]
```diff
@@ -71,10 +83,10 @@ extension LaneIndication {
             secondaryIndication = .straightAhead
         } else if !primaryIndication.isDisjoint(with: .rights) && indications.contains(.slightLeft) {
             // No assets for slight or sharp opposite turns.
-            secondaryIndication = .slightLeft
+            secondaryIndication = .left
```
Comment (1ec5) [inline Sources/MapboxNavigation/LaneView.swift:86]: We still don’t have assets for slight or sharp opposite turns, but at least we have assets for opposite turns. A user is more likely to distinguish between a left lane and a left-right lane than between a left lane and a slight left lane, even if the maneuver requires going slight left.

File: Sources/MapboxNavigation/LaneView.swift [hunk 2]
```diff
@@ -36,6 +40,14 @@ extension LaneIndication {
...
         } else if indications.contains(.sharpRight) && maneuverDirection ?? .sharpRight == .sharpRight {
             primaryIndication = .sharpRight
+        } else if indications.contains(.left) && !indications.contains(.sharpLeft) && !indications.contains(.uTurn) && maneuverDirection ?? .sharpLeft == .sharpLeft {
+            primaryIndication = .left
+        } else if indications.contains(.right) && !indications.contains(.sharpRight) && !indications.contains(.uTurn) && maneuverDirection ?? .sharpRight == .sharpRight {
+            primaryIndication = .right
+        } else if !indications.isDisjoint(with: .lefts) && !indications.isDisjoint(with: .rights) && maneuverDirection?.isLeft ?? false {
+            primaryIndication = .left
+        } else if !indications.isDisjoint(with: .lefts) && !indications.isDisjoint(with: .rights) && maneuverDirection?.isRight ?? false {
+            primaryIndication = .right
```
Comment (1ec5) [inline Sources/MapboxNavigation/LaneView.swift:50]: These cases handle lanes with opposite turns, such as left-right. Previously, we’d fall into the big else case below that tries to choose the “least extreme” indication. It’d be a tie between left and right, so that logic would choose left because it happens to come first. This more generic logic accounts for slight and sharp turns too.

File: Sources/MapboxNavigation/LaneView.swift [hunk 3]
```diff
@@ -36,6 +40,14 @@ extension LaneIndication {
             primaryIndication = .sharpLeft
         } else if indications.contains(.sharpRight) && maneuverDirection ?? .sharpRight == .sharpRight {
             primaryIndication = .sharpRight
+        } else if indications.contains(.left) && !indications.contains(.sharpLeft) && !indications.contains(.uTurn) && maneuverDirection ?? .sharpLeft == .sharpLeft {
+            primaryIndication = .left
+        } else if indications.contains(.right) && !indications.contains(.sharpRight) && !indications.contains(.uTurn) && maneuverDirection ?? .sharpRight == .sharpRight {
+            primaryIndication = .right
```
Comment (1ec5) [inline Sources/MapboxNavigation/LaneView.swift:46]: This is tricky. A distinction between sharp left, left, and left doesn’t matter much in general – a sharp left maneuver should bias the ranking towards whatever is on the left. However, a user is likely to distinguish between a left and a U-turn to the left.

File: Tests/MapboxNavigationTests/LaneViewTests.swift [hunk 4]
```diff
@@ -116,71 +116,75 @@ class LaneViewTests: XCTestCase {
...
     }
     
     func testLaneConfiguration() {
         XCTAssertEqual(LaneConfiguration(rankedIndications: .init(primary: .uTurn, secondary: nil, tertiary: nil), drivingSide: .right),
                        .uTurn(side: .left))
         XCTAssertEqual(LaneConfiguration(rankedIndications: .init(primary: .uTurn, secondary: nil, tertiary: nil), drivingSide: .left),
                        .uTurn(side: .right))
-        XCTAssertNil(LaneConfiguration(rankedIndications: .init(primary: .uTurn, secondary: .left, tertiary: nil), drivingSide: .right))
+        XCTAssertEqual(LaneConfiguration(rankedIndications: .init(primary: .uTurn, secondary: .left, tertiary: nil), drivingSide: .right),
+                       .turnOrUTurn(side: .left, turn: false, uTurn: true))
```
Comment (1ec5) [inline Tests/MapboxNavigationTests/LaneViewTests.swift:168]: These cases simply weren’t updated to reflect the assets added in #2882.

**Comment by 1ec5** [comment]
As an aside, I wonder if we could streamline a lot of this logic by making the maneuver directions comparable. Then we could map a lane’s indications into a range of maneuver directions and compare against that range instead of individual indications.

**Review by S2Ler — APPROVED** [review]
It is hard for me to verify the correctness of this logic, but overall it looks good. Inline comments helped as well.
